src/Chapter3.hs

```diff
@@ -374,6 +382,34 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+-- I decided to add a `1` prefix to these data structures because I repeated them in the last excercise of this chapter
+-- also, the two data structures could also be unified, using a phantom type parameter to distinguish them
+data Knight1 = Knight1 {
+  knight1Health :: Int,
+  knight1Attack :: Int,
+  knight1Gold :: Int
+} deriving Show
+
+data Monster1 = Monster1 {
+  monster1Health :: Int,
+  monster1Attack :: Int,
+  monster1Gold :: Int
+} deriving Show
+
+fight :: Knight1 -> Monster1 -> Int
+fight knight@(Knight1 kHealth kAttack kGold) monster@(Monster1 mHealth mAttack mGold) =  -- RecordWildcards, maybe?
+  if monster1Health newMonster > 0
+    then
+      if knight1Health newKnight <= 0
+      then -1
+      else kGold
+  else
+    kGold + mGold
```

Comment:
`RecordWildCards` are excellent for this task :top: 
You can also use guards to simplify the implementation

Reply:
That's a good hint. I'll post a new solution using guards in a minute. This one's a bit more readable I think